### DISCUSSION TITLE
Could running 6.5.2 with Intel Iris Xe iGPU cause system freeze?

### DISCUSSION THREAD

**Discussion by Jefferydo** [pixijs/pixijs#9137] [Q&A]
Hi! I host a Foundry VTT game that was plagued by hourly freezes until I isolated the culprit as a community module heavily leveraging pixi.js for animation: [https://github.com/Feu-Secret/Tokenmagic](https://github.com/Feu-Secret/Tokenmagic). I tested the module in isolation, with only it and the core app running, and 60 simple webp assets on the canvas.

The Foundry app itself is running pixi.js version 6.5.2. Could the old version be responsible for the freezes?

**Comment by bigtimebuddy** [comment]
Have you tried memory profiling to see if there's a memory leak? Freezes and crashes, in my experience, are usually caused by memory mismanagement.

I don't know of specific issues with 6.5.2, but you can checkout the [releases](https://github.com/pixijs/pixijs/releases) to see newer 6.x version or see a [comparison](https://github.com/pixijs/pixijs/compare/v6.5.2...v6.5.9) of commits between 6.5.2-6.5.9 to see if there might be anything impacting your code.

**Reply by Jefferydo** [reply]
Thanks. I found PixiJS Bunny Mark after posting and my iGPU has no problem with 100,000 of the critters. 

My first suspicion was a memory leak, and I did actually spend an evening with Chrome's performance tab and "Performance Insights." I hit the tutorials, but I had to throw in the towel. I lack the required knowledge.
